### ISSUE TITLE
Non-invasive mode: unpack and drop a chroot script in WSL/Ubuntu

### ISSUE THREAD

**Issue by Artoria2e5** [AOSC-Dev/WSAOSC#1]

**Comment by LER0ever** [comment]
That could be a perfect solution for multiple distribution under WSL, but unfortunately, `chroot` is not yet implemented by Microsoft under WSL

**Comment by Artoria2e5** [comment]
Implemented on my b14959. Any more questions?

**Comment by MingcongBai** [comment]
> Implemented on my b14959. Any more questions?

I would suppose that this is not yet available in the stable version? https://en.wikipedia.org/wiki/Windows_10

**Comment by Artoria2e5** [comment]
Since WSL is being actively updated, to do useful things with it you should use an Insider release for Windows 10.

**Comment by MingcongBai** [comment]
> Since WSL is being actively updated, to do useful things with it you should use an Insider release for Windows 10.

Right, you do have a point. I guess we could run an extra check to see if a "chroot" installation mode is possible to be used? @LER0ever

**Comment by LER0ever** [comment]
OK, I'll add an extra option that shows after the alt-click action. But firstly I have to figure out which build exactly is the chroot function added to WSL.

**Comment by MingcongBai** [comment]
> OK, I'll add an extra option that shows after the alt-click action. But firstly I have to figure out which build exactly is the chroot function added to WSL.

Look above.

**Comment by Artoria2e5** [comment]
Should be there as of 14951. chroot is required for `sshd`, so it is kinda fundamental.

**Comment by LER0ever** [comment]
> Look above.

I know it's implemented in b14959, but it works also fine on my samsung with b14955.
So I need to know the very first build that supports chroot.

**Comment by Artoria2e5** [comment]
Anyway, just don't worry about writing system requirements. You can always use `chroot /` to test...

**Comment by MingcongBai** [comment]
> Should be there as of 14951. chroot is required for sshd, so it is kinda fundamental.

I thought SSH Daemon can work withou chroot functionality...

**Comment by MingcongBai** [comment]
> Anyway, just don't worry about writing system requirements. You can always use chroot / to test...

Nice.
